fhir-search/src/main/java/com/ibm/fhir/search/compartment/CompartmentUtil.java

```diff
@@ -136,7 +136,10 @@ private CompartmentUtil() {
                     for (Resource resource : compartmentDefinition.getResource()) {
                         String inclusionCode = resource.getCode().getValue();
                         List<com.ibm.fhir.model.type.String> params = resource.getParam();
-                        compartmentDefinitionCache.add(inclusionCode, params);
+                        // Make sure to only add the valid resource types (at least with one inclusion) instead of all types.
+                        if (!params.isEmpty()) {
+                            compartmentDefinitionCache.add(inclusionCode, params);
```

Comment:
how is this included in your PR? is this from a rebase? Trying to figure out how it stitched with the other changes.

Reply:
I found this minor issue in my debug, without this change, all resource types will be regards as compartment resource types of any resource type(including itself).